app/api/common.v2/contracts/Sink.ts

```diff
@@ -0,0 +1,6 @@
+interface Sink<T> {
```

Comment:
I like the concept, I'm not sure about the name. I would attach to SaveStream, the same way you named the implementation, factory method and variables.

Reply:
Agree with you